crates/webidl/src/util.rs

```diff
@@ -173,192 +58,344 @@ fn result_ty(t: syn::Type) -> syn::Type {
+            ret = Some(ret.map_or_else(|| result_ty(unit_ty()), |ret| result_ty(ret)))
+        }
+
+        let shim = {
+            let ns = match kind {
+                backend::ast::ImportFunctionKind::Normal => "",
+                backend::ast::ImportFunctionKind::Method { ref class, .. } => class,
+            };
+
+            raw_ident(&format!("__widl_f_{}_{}", rust_name, ns))
+        };
+
+        Some(backend::ast::ImportFunction {
+            function: backend::ast::Function {
+                name,
+                arguments,
+                ret,
+                rust_attrs: vec![],
+                rust_vis: public(),
+            },
+            rust_name,
+            js_ret,
+            catch,
+            structural,
+            kind,
+            shim,
+        })
+    }
+
+    pub fn create_basic_method(
+        &self,
+        arguments: &[webidl::ast::Argument],
+        name: Option<&String>,
+        return_type: &webidl::ast::ReturnType,
+        self_name: &str,
+        is_static: bool,
+        catch: bool,
+    ) -> Option<backend::ast::ImportFunction> {
+        let name = match name {
             None => {
-                warn!("Operation's return type is not yet supported: {:?}", ty);
+                warn!("Operations without a name are unsupported");
+                return None;
+            }
+            Some(ref name) => name,
+        };
+
+        let kind = backend::ast::ImportFunctionKind::Method {
+            class: self_name.to_string(),
+            ty: ident_ty(rust_ident(self_name.to_camel_case().as_str())),
+            kind: backend::ast::MethodKind::Operation(backend::ast::Operation {
+                is_static,
+                kind: backend::ast::OperationKind::Regular,
+            }),
+        };
+
+        let ret = match return_type {
+            webidl::ast::ReturnType::Void => None,
+            webidl::ast::ReturnType::NonVoid(ty) => {
+                match self.webidl_ty_to_syn_ty(ty, TypePosition::Return) {
+                    None => {
+                        warn!("Operation's return type is not yet supported: {:?}", ty);
+                        return None;
+                    }
+                    Some(ty) => Some(ty),
+                }
+            }
+        };
+
+        self.create_function(
+            &name,
+            arguments
+                .iter()
+                .map(|arg| (&*arg.name, &*arg.type_, arg.variadic)),
+            ret,
+            kind,
+            false,
+            catch,
+        )
+    }
+
+    pub fn create_getter(
+        &self,
+        name: &str,
+        ty: &webidl::ast::Type,
+        self_name: &str,
+        is_static: bool,
+        is_structural: bool,
+        catch: bool,
+    ) -> Option<backend::ast::ImportFunction> {
+        let ret = match self.webidl_ty_to_syn_ty(ty, TypePosition::Return) {
+            None => {
+                warn!("Attribute's type does not yet support reading: {:?}", ty);
```

Comment:
Should this warning message s/reading/returning/ ?

Reply:
I guess this is how it was before, but the message no longer makes sense to me, and I assume I originally wrote it...